spring-integration-file/src/main/java/org/springframework/integration/file/remote/gateway/AbstractRemoteFileOutboundGateway.java

```diff
@@ -868,13 +875,16 @@ else if (FileExistsMode.IGNORE != fileExistsMode) {
 			return mGetWithRecursion(message, session, remoteDirectory, remoteFilename);
 		}
 		else {
+			remoteDirectory = (remoteDirectory != null
+					? remoteDirectory
+					: "." + this.remoteFileTemplate.getRemoteFileSeparator());
```

Comment:
Yeah! I got it. We should convert `null` to `./` for the next recursion level, but not on the root.
Fixing...